Common/Data/Custom/PsychSignal/PsychSignalSentiment.cs

```diff
@@ -29,6 +29,26 @@ namespace QuantConnect.Data.Custom.PsychSignal
     /// </summary>
     public class PsychSignalSentiment : BaseData
     {
+        /// <summary>
+        /// The EndTime of the bar represents the time the data ended and
+        /// the time it should be emitted in backtesting/live trading
+        /// </summary>
+        public override DateTime EndTime {
+            get { return Time + Period + FixedLiveOffset; }
+            set { Time = value - Period; }
+        }
+
+        /// <summary>
+        /// Time from the start of the bar until the end of the bar per each data point
+        /// </summary>
+        public TimeSpan Period { get; set; } = TimeSpan.FromMinutes(1);
+
+        /// <summary>
+        /// Time it takes for data to become available on the PsychSignal API after
+        /// the time advances to the next minute
+        /// </summary>
+        public static TimeSpan FixedLiveOffset { get; set; } = TimeSpan.FromSeconds(0);
```

Comment:
¿Can you please explain this property?

Reply:
In live trading the bars appear 15s after the minute; we should make backtesting model this as closely as possible. We could set it to 15s now so backtesting accurately modeled...